hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/TestRMWebServicesCapacitySchedDynamicConfig.java

```diff
@@ -18,311 +18,161 @@
+    conf.put("yarn.scheduler.capacity.root.queues", "default, test1, test2");
+    conf.put("yarn.scheduler.capacity.root.test1.queues", "test1_1, test1_2, test1_3");
+    conf.put("yarn.scheduler.capacity.root.default.capacity", "4w");
+    conf.put("yarn.scheduler.capacity.root.test1.capacity", "16w");
+    conf.put("yarn.scheduler.capacity.root.test2.capacity", "12w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_1.capacity", "2w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_2.capacity", "2w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_3.capacity", "12w");
+    MockRM rm = createMutableRM(createConfiguration(conf));
+    runTest(EXPECTED_FILE_TMPL, "testWeightMode", rm, resource());
   }
 
   @Test
-  public void testSchedulerResponseWeightModeWithAutoCreatedQueues()
-      throws Exception {
-    Configuration config = CSConfigGenerator
-        .createWeightConfigWithAutoQueueCreationEnabled();
-    config.setInt(CapacitySchedulerConfiguration
-        .getQueuePrefix("root.autoParent1") +
-        AutoCreatedQueueTemplate.AUTO_QUEUE_TEMPLATE_PREFIX +
-        "maximum-applications", 300);
-
-    initResourceManager(config);
-    initAutoQueueHandler(1200 * GB);
-
-    // same as webapp/scheduler-response-WeightMode.json, but with effective resources filled in
-    assertJsonResponse(sendRequest(),
-        "webapp/scheduler-response-WeightModeWithAutoCreatedQueues-Before.json");
-
-    //Now create some auto created queues
-    createQueue("root.auto1");
-    createQueue("root.auto2");
-    createQueue("root.auto3");
-    createQueue("root.autoParent1.auto4");
-    createQueue("root.autoParent2.auto5");
-    createQueue("root.parent.autoParent3.auto6");
-
-    assertJsonResponse(sendRequest(),
-        "webapp/scheduler-response-WeightModeWithAutoCreatedQueues-After.json");
-  }
-
-  private void initAutoQueueHandler(int nodeMemory) throws Exception {
-    CapacityScheduler cs = (CapacityScheduler) rm.getResourceScheduler();
-    autoQueueHandler = cs.getCapacitySchedulerQueueManager();
-    rm.registerNode("h1:1234", nodeMemory); // label = x
+  public void testWeightModeFlexibleAQC() throws Exception {
+    Map<String, String> conf = new HashMap<>();
+    conf.put("yarn.scheduler.capacity.root.queues", "default, test1, test2");
+    conf.put("yarn.scheduler.capacity.root.test1.queues", "test1_1, test1_2, test1_3");
+    conf.put("yarn.scheduler.capacity.root.default.capacity", "4w");
+    conf.put("yarn.scheduler.capacity.root.test1.capacity", "16w");
+    conf.put("yarn.scheduler.capacity.root.test2.capacity", "12w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_1.capacity", "2w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_2.capacity", "2w");
+    conf.put("yarn.scheduler.capacity.root.test1.test1_3.capacity", "12w");
+
+    Configuration config = createConfiguration(conf);
+    setupAQC(config, "yarn.scheduler.capacity.root.test1.");
```

Comment:
Too many queues are created, hard to reason about the expected JSON.

How to make it better:
 - enable AQC only on one queue (before this change it was the root queue, we can use root.test1 there is 16GB and 16VCore to share)
 - create the queues only on one queue
 - the effectiveMin resources should make sense on the newly created queues
 - the applied template configs should be easily checked, the acl usernames should be meaningful for this to work
      - auto-queue-creation-v2.parent-template.acl_                            {parentUser, parentAdmin}
      - autoParent1.auto-queue-creation-v2.leaf-template.acl_           {ap1User, ap2Admin}
      - *.auto-queue-creation-v2.leaf-template.acl_                              {leafUser, leafAdmin}
      - parent.*.auto-queue-creation-v2.leaf-template.acl                   {pLeafUser, pLeafAdmin}
 
 Something like this:     
```
      root.test_1.auto1 [2gb 2vcore]   {leafUser, leafAdmin}
      root.test_1.auto2 [2gb 2vcore]  {leafUser, leafAdmin}
      root.test_1.autoParent1.auto3 [1gb 1 vcore]  {parentUser, parentAdmin} {ap1User, ap1Admin}
      root.test_1.autoParent1.auto4 [1gb 1vcore] {parentUser, parentAdmin} {ap1User, ap1Admin}
      root.test_1.autoParent2.auto5 [1gb 1 vcore] {parentUser, parentAdmin} {leafUser, leafAdmin}
      root.test_1.parent.autoParent2.auto6 [2gb 2 vcore] {parentUser, parentAdmin} {pLeafUser, pLeafAdmin}
      root.test_1.parent2.auto7 [2gb 2 vcore] {parentUser, parentAdmin} {leafUser, leafAdmin}
      root.test_1.parent3.auto8 [2gb 2 vcore] {parentUser, parentAdmin} {leafUser, leafAdmin}
```